### PR TITLE
Feature/ml prediction pipeline

### PR DESCRIPTION
**Pull request by fivejjs**
The code is to create the crop-mask from semiannual geomedia-mad .

### CODE REVIEW

File: libs/stats/odc/stats/_gm_ml_pred.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+        return self.bands
+
+    def input_data(self, task: Task) -> xr.Dataset:
+        """
+        assemble the input data and do prediction here.
+        This method work as pipeline
+        """
+        dc = Datacube(app=self.target_product)
+
+        geobox = self.africa_N.tile_geobox(task.tile_index)
```
Comment (Kirill888) [inline libs/stats/odc/stats/_gm_ml_pred.py:508]: you don't need this, task object already includes `.geobox` property

File: libs/stats/odc/stats/_gm_ml_pred.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+    VERSION = "0.1.0"
+    PRODUCT_FAMILY = "geomedian"
+
+    source_product = 'gm_s2_semiannual'
+    target_product = 'crop_mask_eastern'
+
+    def __init__(self):
+        # target band to be saved
+        self.bands = ('mask', 'prob')  # skip, 'filtered')
+        self.africa_N = GRIDS[f'africa_{PredConf.resolution[1]}']
```
Comment (Kirill888) [inline libs/stats/odc/stats/_gm_ml_pred.py:495]: no, grid comes from Task.

File: libs/stats/odc/stats/_gm_ml_pred.py [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+        s3_bucket, s3_key = path.split('/')[2], path.split('/')[3:]
+        s3_key = '/'.join(s3_key)
+        with BytesIO() as f:
+            boto3.client("s3", config=Config(signature_version=UNSIGNED)).download_fileobj(Bucket=s3_bucket, Key=s3_key,
+                                                                                           Fileobj=f)
+            f.seek(0)
+            model = joblib.load(f)
+
+    # Path is a local directory 
+    else:
```
Comment (Kirill888) [inline libs/stats/odc/stats/_gm_ml_pred.py:219]: Please use `s3_download` or `s3_open` from here

https://github.com/opendatacube/odc-tools/blob/3e9c2a42a33cf5bf5b82553230685fee74061e70/libs/aws/odc/aws/__init__.py#L308-L362

File: libs/stats/odc/stats/_gm_ml_pred.py [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+    """
+    down scale band not in exclude list.
+    :param ds:
+    :param exclude:
+    :param scale:
+    :return:
+    """
+    for band in ds.data_vars:
+        if band not in exclude:
+            ds[band] = ds[band] / scale
```
Comment (Kirill888) [inline libs/stats/odc/stats/_gm_ml_pred.py:339]: not sure it's a good idea to modify `ds` object in place.

File: libs/stats/odc/stats/_gm_ml_pred.py (hunk 3 above)
Comment (fivejjs) [reply libs/stats/odc/stats/_gm_ml_pred.py:219]: nice.

File: libs/stats/odc/stats/_gm_ml_pred.py (hunk 4 above)
Comment (fivejjs) [reply libs/stats/odc/stats/_gm_ml_pred.py:339]: I will discuss it with Chad

**Comment by Kirill888** [comment]
@fivejjs please cleanup git history, there are too many "fixup" commits currently. It's ok to "force push" to your own branch.

**Pull request closed without merging**